components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/GroupManagementV2AuditLogger.java

```diff
@@ -0,0 +1,157 @@
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.isEnableV2AuditLogs;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.jsonObjectToMap;
+import static org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils.triggerAuditLogEvent;
+import static org.wso2.carbon.user.mgt.listeners.utils.ListenerUtils.getInitiatorId;
+
+/**
+ * This v2 audit logger logs the Group Management success activities.
+ */
+public class GroupManagementV2AuditLogger extends AbstractIdentityGroupOperationEventListener {
+
+    public boolean isEnable() {
+
+        if (super.isEnable()) {
+            return isEnableV2AuditLogs();
+        }
+        return false;
+    }
+
+    @Override
+    public boolean postAddGroup(String groupName, String groupId, List<String> userIds, List<Claim> claims,
+                                UserStoreManager userStoreManager) {
+
+        if (!isEnable()) {
+            return true;
+        }
+        JSONObject dataObject = new JSONObject();
+        if (CollectionUtils.isNotEmpty(userIds)) {
+            dataObject.put(USERS_FIELD, new JSONArray(userIds));
+        }
+        dataObject.put(GROUP_NAME_FIELD, groupName);
+        AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(getInitiatorId(),
+                LoggerUtils.getInitiatorType(getInitiatorId()), groupId, Target.Group.name(), ADD_GROUP_ACTION)
+                .data(jsonObjectToMap(dataObject));
+        triggerAuditLogEvent(auditLogBuilder);
+        return true;
+    }
+
+    @Override
+    public boolean postDeleteGroup(String groupId, String groupName, UserStoreManager userStoreManager) {
+
+        if (!isEnable()) {
+            return true;
+        }
+        AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(getInitiatorId(),
+                LoggerUtils.getInitiatorType(getInitiatorId()), groupId, Target.Group.name(), DELETE_GROUP_ACTION);
+        triggerAuditLogEvent(auditLogBuilder);
+        return true;
+    }
+
+    @Override
+    public boolean postRenameGroup(String groupId, String newGroupName, UserStoreManager userStoreManager) {
+
+        if (!isEnable()) {
+            return true;
+        }
+        JSONObject dataObject = new JSONObject();
+        dataObject.put(GROUP_NAME_FIELD, newGroupName);
+        AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(getInitiatorId(),
+                LoggerUtils.getInitiatorType(getInitiatorId()), groupId, Target.Group.name(), UPDATE_GROUP_NAME_ACTION)
+                .data(jsonObjectToMap(dataObject));
+        triggerAuditLogEvent(auditLogBuilder);
+        return true;
+    }
+
+    @Override
+    public boolean postUpdateUserListOfGroup(String groupId, List<String> deletedUserIds, List<String> newUserIds,
+                                             UserStoreManager userStoreManager) {
+
+        if (!isEnable()) {
+            return true;
```

Comment:
check the previous comment

Reply:
https://github.com/wso2/carbon-identity-framework/pull/5593#discussion_r1538620086